libminifi/include/FlowController.h

```diff
@@ -64,8 +64,9 @@ class FlowController : public core::CoreComponent {
   /**
    * Flow controller constructor
    */
-  FlowController(std::shared_ptr<core::Repository> provenance_repo,
-                 std::shared_ptr<core::Repository> flow_file_repo,
+  FlowController(const std::shared_ptr<core::Repository> &provenance_repo,
```

Comment:
we could move the original std shared ptr to subsume ownership, but I'm not sure we want that. We want to say we're sharing ownership with this object and thus this object ( FlowController ) helps in determining the lifetime of shared_ptr.